[AX] Stop soft-linking libAccessibility.dylib in WebCore

ddkilzer opened the pull request:
#### 82c4bdb4701baf19e7f7ff6e70312ec1bb1e5afd
<pre>
[AX] Stop soft-linking libAccessibility.dylib in WebCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=271953">https://bugs.webkit.org/show_bug.cgi?id=271953</a>
&lt;<a href="https://rdar.apple.com/125689689">rdar://125689689</a>&gt;

Reviewed by NOBODY (OOPS!).

libAccessibility.dylib is currently weak-linked from WebCore.framework,
so there is no need to soft-link it for a single symbol.

The original need to weak-link libAccessibility.dylib was because it
wasn&apos;t available on Recovery OS, but that changed in macOS 13 Ventura.
However, weak-linking is being kept since it may prevent a dependency
cycle during OS builds.

No tests since there are no testable changes in behavior.

* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
- Remove AccessibilitySupportSoftLink.{cpp,h} files.
* Source/WebCore/PAL/pal/cocoa/UsageTrackingSoftLink.h:
(_AXSIsolatedTreeModeFunctionIsAvailable): Remove unneeded macro.
* Source/WebCore/PAL/pal/spi/cocoa/AccessibilitySupportSoftLink.cpp: Remove.
* Source/WebCore/PAL/pal/spi/cocoa/AccessibilitySupportSoftLink.h: Remove.
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.cpp:
- Change #import to #include for C++ source.
(WebCore::AXIsolatedObject::children):
- Remove missing symbol check for _AXSIsolatedTreeMode().
* Source/WebCore/accessibility/isolatedtree/mac/AXIsolatedObjectMac.mm:
- Remove unused header.
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
(WebCore::AXObjectCache::isIsolatedTreeEnabled):
(WebCore::AXObjectCache::initializeAXThreadIfNeeded):
- Remove missing symbol check for _AXSIsolatedTreeMode().
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
- Remove unused header.
* Tools/TestWebKitAPI/Tests/WebKit/AccessibilityIncreaseContrast.mm:
- Remove unused header.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/82c4bdb4701baf19e7f7ff6e70312ec1bb1e5afd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47594 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26780 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50264 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50276 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43643 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/49901 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32552 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24241 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38752 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48176 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WK2-Tests-EWS "Waiting to run tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20051 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/42210 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5636 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [⏳ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-WK2-Tests-EWS "Waiting to run tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52160 "Built successfully") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22628 "Build is in progress. Recent messages:OS: Sonoma (14.4.1), Xcode: 15.3; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/18968 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/46058 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23901 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45084 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24690 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23620 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (approved):
It would be good to check a build of this with `otool -L` and make sure that libAccessibility is indeed weak-linked in all the relevant binaries. As I understand it, a single use of the library through a strong declaration will cause the linker to record the runtime dependency as strong. Though I can't see how any of WebKit's uses might have slipped through this PR since the declaration headers are gone.

Since this is still a draft, please let me know if you plan to change it substantially.

Commit:
[AX] Stop soft-linking libAccessibility.dylib in WebCore
https://bugs.webkit.org/show_bug.cgi?id=271953
<rdar://125689689>

Reviewed by NOBODY (OOPS!).

libAccessibility.dylib is currently weak-linked from WebCore.framework,
so there is no need to soft-link it for a single symbol.

The original need to weak-link libAccessibility.dylib was because it
wasn't available on Recovery OS, but that changed in macOS 13 Ventura.
However, weak-linking is being kept since it may prevent a dependency
cycle during OS builds.

No tests since there are no testable changes in behavior.

* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
- Remove AccessibilitySupportSoftLink.{cpp,h} files.
* Source/WebCore/PAL/pal/cocoa/UsageTrackingSoftLink.h:
(_AXSIsolatedTreeModeFunctionIsAvailable): Remove unneeded macro.
* Source/WebCore/PAL/pal/spi/cocoa/AccessibilitySupportSoftLink.cpp: Remove.
* Source/WebCore/PAL/pal/spi/cocoa/AccessibilitySupportSoftLink.h: Remove.
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.cpp:
- Change #import to #include for C++ source.
(WebCore::AXIsolatedObject::children):
- Remove missing symbol check for _AXSIsolatedTreeMode().
* Source/WebCore/accessibility/isolatedtree/mac/AXIsolatedObjectMac.mm:
- Remove unused header.
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
(WebCore::AXObjectCache::isIsolatedTreeEnabled):
(WebCore::AXObjectCache::initializeAXThreadIfNeeded):
- Remove missing symbol check for _AXSIsolatedTreeMode().
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
- Remove unused header.
* Tools/TestWebKitAPI/Tests/WebKit/AccessibilityIncreaseContrast.mm:
- Remove unused header.

emw-apple reviewed (commented):
> However, weak-linking is being kept since it may prevent a dependency
cycle during OS builds.

How does weak linking help break the dependency cycle? IIUC, weak linking has the same SDK dependencies at build time, so this change makes WebCore load libAccessibility.tbd at build time and pick it up as a dependency.

Alimsayed reviewed (commented):
https://bugs.webkit.org/show_bug.cgi?id=271953
